### DISCUSSION TITLE
Support for alternate methods for linux capture (wayland and more) support, proposals included.

### DISCUSSION THREAD

**Discussion by Quackdoc** [rustdesk/rustdesk#10300] [Ideas]
Portals is not great, It's support is lackluster and can be fairly buggy, It would be nice to support other capture methods. The issue is that these methods may not work in flatpak. However outside of flatpak they can make for a much more ergonomic experience allowing for things like unattended access.

WAYLAND:

[wlr-screencopy-unstable-v1](https://wayland.app/protocols/wlr-screencopy-unstable-v1): One of the current most popular capture protocols. It has good support and would be immediately usable for a good number of compositors, but ext-image-capture-source has replaced it.

[ext-image-capture-source](ttps://wayland.app/protocols/ext-image-capture-source-v1) These are the upstream wayland protocols for video capture. While these don't currently have great widespread support. They are the official ones and should be the "proper" future solution https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/124

GENERIC:

Direct pipewire capture: This may be the easiest to do. As rustdesk already has pipewire capture support Being able to feed a pipewire source directly into rustdesk would mean minimal work and would also be the most flexible. While not useful in isolation, It can be paired with additional services/backends which can make it more modular. 

kmsgrab: kmsgrab is one of the more reliable forms of capture but requires additional permissions. It will work on most hardware across most DE's but some may still have issues.

Personally I think direct pipewire support would be really nice. It can easily be extended to support most desktops using kmsgrab, or more specific tasks such as having a dedicated remote desktop device addon device (raspberry pi for instance)

**Comment by rustdesk** [comment]
PR is welcome

**Comment by lin-calvin** [comment]
i think rdp is a good idea, connect with the DE's rdp server from the rustdesk server side. And we can ignore all underlying captureing method implements

**Reply by xlionjuan** [reply]
I think not

**Comment by xlionjuan** [comment]
https://github.com/rustdesk/rustdesk/discussions/10016#discussioncomment-11757252

> Wayland doesn't require a confirmation dialog, but it is implemented that way so that the mechanism to require a dialog exists. There's nothing stopping anyone from making a portal that doesn't and automatically grants access, I'm just not sure if anyone has such a portal implemented yet.
> Cage is based on wlroots, and xdg-desktop-portal-wlr seems to have a config file where you can set `chooser_type = none` which will unconditionally grant access which seems to be what you're looking for.

**Comment by xlionjuan** [comment]
# XDG Portal Pre-Authorization

## Pre-authorize applications to allow usage of XDG portal tech without interactive dialogs

Plasma 6.3

https://develop.kde.org/docs/administration/portal-permissions/
